tests/devices_test/dependencies_test.py

```diff
@@ -129,3 +134,111 @@ def test_dependencies_handling(self):
+                os.unlink(fn)
+
+    def test_weak_dependencies(self):
+
+        self.bvt = blivet.Blivet()
+        # reinitialize blockdev without the plugins
+        # TODO: uncomment (workaround (1/3) for blivet.reset fail)
+        # blivet.blockdev.try_reinit(require_plugins=[], reload=True)
+
+        self.bvt.disk_images["disk1"] = self.disk1_file
+
+        try:
+            self.bvt.reset()
+        except blivet.blockdev.BlockDevNotImplementedError:  # pylint: disable=catching-non-exception
+            self.fail("Improper handling of missing libblockdev plugin")
+
+        # TODO: remove line (workaround (2/3) for blivet.reset fail)
+        blivet.blockdev.try_reinit(require_plugins=[], reload=True)
+
+        disk1 = self.bvt.devicetree.get_device_by_name("disk1")
+        self.bvt.initialize_disk(disk1)
+        pv = self.bvt.new_partition(size=Size("8GiB"), fmt_type="lvmpv")
+        btrfs = self.bvt.new_partition(size=Size("1GiB"), fmt_type="btrfs")
+
+        # TODO: remove "with" (workaround (3/3) for blivet.reset fail)
+        with patch.object(StorageDevice, "unavailable_dependencies",
+                          new_callable=PropertyMock(return_value=['dummy'])):
+            with six.assertRaisesRegex(self, ValueError, "requires unavailable_dependencies"):
+                self.bvt.create_device(pv)
+            with six.assertRaisesRegex(self, ValueError, "requires unavailable_dependencies"):
+                self.bvt.create_device(btrfs)
+
+        with patch.object(StorageDevice, "unavailable_dependencies",
+                          new_callable=PropertyMock(return_value=set())):
+            try:
+                self.bvt.create_device(pv)
+                self.bvt.create_device(btrfs)
+            except blivet.blockdev.BlockDevNotImplementedError:  # pylint: disable=catching-non-exception
+                self.fail("Improper handling of missing libblockdev plugin")
+
+        try:
+            vg = self.bvt.new_vg(parents=[pv])
+        except blivet.blockdev.BlockDevNotImplementedError:  # pylint: disable=catching-non-exception
+            self.fail("Improper handling of missing libblockdev plugin")
+        self.bvt.create_device(vg)
+
+        try:
+            lv1 = self.bvt.new_lv(fmt_type="ext4", size=Size("1GiB"), parents=[vg])
+            lv2 = self.bvt.new_lv(fmt_type="ext4", size=Size("1GiB"), parents=[vg])
+        except blivet.blockdev.BlockDevNotImplementedError:  # pylint: disable=catching-non-exception
+            self.fail("Improper handling of missing libblockdev plugin")
+        self.bvt.create_device(lv1)
+        self.bvt.create_device(lv2)
+
+        try:
+            pool = self.bvt.new_lv_from_lvs(vg, name='pool', seg_type="thin-pool", from_lvs=[lv1, lv2])
+        except blivet.blockdev.BlockDevNotImplementedError:  # pylint: disable=catching-non-exception
+            self.fail("Improper handling of missing libblockdev plugin")
+        self.bvt.devicetree._add_device(pool)
```

Comment:
`Blivet.create_device` is what you want -- it schedules an action to create the device and will add the new device to the tree. You should see an exception when it schedules the action if the plugin is missing.